src/main/java/com/opentable/kafka/builders/KafkaBaseBuilder.java

```diff
@@ -148,6 +150,10 @@ void withLogging(boolean enabled) {
     void withSamplingRatePer10Seconds(final int rate) {
         loggingSampleRate = rate;
     }
+    void withSamplingRate(final int rate) {
```

Comment:
This is a bit confusing. Two similar methods, etc. Rename to withRandomSamplingRate. Also the withSamplingRate shhould reverse the type back.